File: OldSnacks/student_grade.py



number_student = int(input("How many students do you have?: "))
number_subject = int(input("How many subjects do they offer?: "))

for student in range(1, number_student + 1):
    print(f"\nEntering scores for Student {student}")
    
    for subject in range(1, number_subject + 1):
       
        while True:
            score = int(input(f"  Enter score for Subject {subject}: "))
            if 0 <= score <= 100:
                break
            else:
                print("  Invalid score. Please enter a number between 0 and 100.")
    
 
"""
score += 1
if score < 0 and score <= 100:
        print("invalid score")
        
total = 0
average = 0

#for number in number_student and number_subject:

def calculating_total():
	total += score

def calculating_average():

	average = total/ number_subject
    
"""

File: Java/StudentGrade.java
import java.util.Scanner;

public class StudentGrade {

    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);

        System.out.print("Enter the number of students: ");
        int numberStudent = input.nextInt();

        System.out.print("Enter the number of subjects: ");
        int numberSubject = input.nextInt();

        int[][] scores = new int[numberStudent][numberSubject];
	

        for (int i = 0; i < numberStudent; i++) {
            System.out.println("\nEntering scores for Student:" + (i + 1) + " : ");
		
	
            for (int j = 0; j < numberSubject; j++) {
		int score;
		while(true){
                System.out.print("Enter score for Subject: " + (j + 1) + " : ");
                score = input.nextInt(); 
		
		if(score >= 0 && score <= 100){
			scores[i][j] = score;
			break;
		} else{
		System.out.println("invalid number. enter scores from 0 - 100: ");
		}

            }
        }
}

        System.out.println("\nStudent Scores, Totals, and Averages:");
        for (int i = 0; i < numberStudent; i++) {
            int total = 0;
           
	 System.out.print("Student " + (i + 1) + " scores: ");
            for (int j = 0; j < numberSubject; j++) {
                System.out.print(scores[i][j] + " ");
                total += scores[i][j];
            }
            double average = (double) total / numberSubject;
            System.out.printf("    Total: %d   Average: %.2f%n", total, average);
	        }




   
    }
}
